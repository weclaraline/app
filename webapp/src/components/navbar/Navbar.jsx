import React from 'react';
import {
  AppBar,
  Toolbar,
  makeStyles,
  createStyles,
  Typography,
} from '@material-ui/core';
import { NavLink } from 'react-router-dom';

import Notifications from './Notifications';
import Avatar from './Avatar';
import Logo from '../../assets/logo.svg';

const useStyles = makeStyles((theme) =>
  createStyles({
    toolbar: {
      flexWrap: 'wrap',
    },
    logo: {
      marginRight: '50px',
      width: '40px',
      [theme.breakpoints.down('md')]: {
        marginRight: 'auto',
      },
    },
    navigation: {
      display: 'flex',
      marginRight: 'auto',
      '& a': {
        color: '#fff',
        textTransform: 'uppercase',
        display: 'block',
        padding: '1.8rem 2.5rem',
        textDecoration: 'none',
        position: 'relative',
        '&.selected:after, &:hover:after': {
          opacity: 1,
        },
        '&:after': {
          content: '""',
          display: 'block',
          position: 'absolute',
          bottom: 0,
          left: 0,
          width: '100%',
          height: '3px',
          backgroundColor: '#fff',
          opacity: 0,
          transition: 'all 0.3s ease-in-out',
        },
      },
      [theme.breakpoints.down('md')]: {
        flex: '0 0 100%',
        order: 2,
        marginTop: '20px',
        '& a': {
          padding: 0,
          '&:first-of-type': {
            marginRight: '20px',
          },
        },
      },
    },
    avatar: {
      marginLeft: '30px',
    },
    actions: {
      display: 'flex',
      [theme.breakpoints.down('md')]: {
        order: 1,
      },
    },
  })
);

const NavBar = () => {
  const classes = useStyles();

  return (
    <AppBar position="relative">
      <Toolbar className={classes.toolbar}>
        <img src={Logo} alt="Weclaraline" className={classes.logo} />
        <nav className={classes.navigation}>
          <NavLink to="/facturas" activeClassName="selected">
            <Typography variant="subtitle1">Facturas</Typography>
          </NavLink>
          <NavLink to="/mis-gastos" activeClassName="selected">
            <Typography variant="subtitle1">Mis gastos</Typography>
          </NavLink>
        </nav>
        <div className={classes.actions}>
          <Notifications />
          <Avatar className={classes.avatar} />
        </div>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
