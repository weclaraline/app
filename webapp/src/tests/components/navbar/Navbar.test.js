import '../../testing_utils';
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Navbar from "../../../components/navbar/Navbar";

it("Renders original content", ()=> {
    const tree = shallow(<Navbar />);
    expect(toJson(tree)).toMatchSnapshot();
});