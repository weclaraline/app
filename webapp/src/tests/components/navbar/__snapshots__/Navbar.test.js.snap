// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders original content 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="relative"
>
  <WithStyles(ForwardRef(Toolbar))
    className="makeStyles-toolbar-1"
  >
    <StarIcon
      className="makeStyles-logo-2"
    />
    <nav
      className="makeStyles-navigation-3"
    >
      <NavLink
        activeClassName="selected"
        to="/facturas"
      >
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Facturas
        </WithStyles(ForwardRef(Typography))>
      </NavLink>
      <NavLink
        activeClassName="selected"
        to="/mis-gastos"
      >
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Mis gastos
        </WithStyles(ForwardRef(Typography))>
      </NavLink>
    </nav>
    <div
      className="makeStyles-actions-5"
    >
      <Notifications />
      <Avatar
        className="makeStyles-avatar-4"
      />
    </div>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
